pytorch_lightning/trainer/callback_hook.py

```diff
@@ -303,7 +302,7 @@ def on_load_checkpoint(self, checkpoint):
             )
 
         for callback in self.callbacks:
-            state = callback_states.get(type(callback))
+            state = callback_states.get(callback.state_id, callback_states.get(callback._legacy_state_id))
```

Comment:
Should `callback_states.get(callback._legacy_state_id)` be changed to `callback_states[callback._legacy_state_id]`

so we don't silently return `None` and instead fail?

We expect the legacy state id to be defined in all cases

Reply:
actually this won't work. We are loading a checkpoint here and we .get() a key but it might be a checkpoint with the new format or the old. So if the checkpoint has legacy keys, `callback_states.get(callback._legacy_state_id)` will return us the right answer, otherwise for new checkpoints above 1.4 we get the element by the new key.